x-pack/plugins/cases/public/components/visualizations/actions/add_to_new_case.tsx

```diff
@@ -0,0 +1,141 @@
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import React, { useEffect, useMemo } from 'react';
+import { unmountComponentAtNode } from 'react-dom';
+import { Router } from 'react-router-dom';
+
+import { createAction } from '@kbn/ui-actions-plugin/public';
+import { isErrorEmbeddable } from '@kbn/embeddable-plugin/public';
+
+import { EuiThemeProvider } from '@kbn/kibana-react-plugin/common';
+import { toMountPoint } from '@kbn/kibana-react-plugin/public';
+import { CommentType } from '../../../../common';
+import {
+  getCaseOwner,
+  getCasePermissions,
+  hasCasePermissions,
+  hasInput,
+  isLensEmbeddable,
+} from './utils';
+import { KibanaContextProvider } from '../../../common/lib/kibana';
+
+import type { ActionContext, CaseUIActionProps, DashboardVisualizationEmbeddable } from './types';
+import CasesProvider from '../../cases_context';
+import { ADD_TO_CASE_SUCCESS, ADD_TO_NEW_CASE_DISPLAYNAME } from './translations';
+import { useCasesAddToNewCaseFlyout } from '../../create/flyout/use_cases_add_to_new_case_flyout';
+
+export const ACTION_ID = 'embeddable_addToNewCase';
+export const DEFAULT_DARK_MODE = 'theme:darkMode' as const;
+
+interface Props {
+  embeddable: DashboardVisualizationEmbeddable;
+  onSuccess: () => void;
+  onClose: () => void;
+}
+
+const AddToNewCaseFlyoutWrapper: React.FC<Props> = ({ embeddable, onClose, onSuccess }) => {
+  const { attributes, timeRange } = embeddable.getInput();
+  const createNewCaseFlyout = useCasesAddToNewCaseFlyout({
+    onClose,
+    onSuccess,
+    toastContent: ADD_TO_CASE_SUCCESS,
+  });
+
+  const attachments = useMemo(
+    () => [
+      {
+        comment: `!{lens${JSON.stringify({
+          timeRange,
+          attributes,
+        })}}`,
+        type: CommentType.user as const,
+      },
+    ],
+    [attributes, timeRange]
+  );
+
+  useEffect(() => {
+    createNewCaseFlyout.open({ attachments });
+  }, [attachments, createNewCaseFlyout]);
+
+  return null;
+};
+
+AddToNewCaseFlyoutWrapper.displayName = 'AddToNewCaseFlyoutWrapper';
+
+export const createAddToNewCaseLensAction = ({
+  core,
+  plugins,
+  storage,
+  history,
+  caseContextProps,
+}: CaseUIActionProps) => {
+  const { application: applicationService, theme, uiSettings } = core;
+  let currentAppId: string | undefined;
+  applicationService?.currentAppId$.subscribe((appId) => {
+    currentAppId = appId;
+  });
+
+  const owner = getCaseOwner(currentAppId);
+  const casePermissions = getCasePermissions(applicationService.capabilities, owner);
+
+  return createAction<ActionContext>({
+    id: ACTION_ID,
+    type: 'actionButton',
+    getIconType: () => 'casesApp',
+    getDisplayName: () => ADD_TO_NEW_CASE_DISPLAYNAME,
+    isCompatible: async ({ embeddable }) =>
+      !isErrorEmbeddable(embeddable) &&
+      isLensEmbeddable(embeddable) &&
+      hasCasePermissions(casePermissions) &&
```

Comment:
We need to check if the user has `create` permissions in this scenario. They don't necessarily need `update` and `read` but there's no current way to give a user only create and not update/read at the moment.